x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_schema.tsx

```diff
@@ -147,19 +158,52 @@ export const EngineSchema: React.FC = () => {
   const [onlyShowConflicts, setOnlyShowConflicts] = useState<boolean>(false);
   const { isLoadingEngineSchema, schemaFields } = useValues(EngineViewLogic);
   const { fetchEngineSchema } = useActions(EngineViewLogic);
+
+  const [isFilterByPopoverOpen, setIsFilterByPopoverOpen] = useState<boolean>(false);
   const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, JSX.Element>>(
     {}
   );
+  // get all the elasticsearch Field Types
+  const esFieldTypes = Object.values(ES_FIELD_TYPES).map((fieldDataTypeName) => ({
+    checked: undefined,
+    label: fieldDataTypeName.toString(),
+  }));
+  const [selectedEsFieldTypes, setSelectedEsFieldTypes] = useState(esFieldTypes);
 
   const toggleOnlyShowConflicts = useCallback(() => {
     setOnlyShowConflicts(!onlyShowConflicts);
     setItemIdToExpandedRowMap({});
   }, [onlyShowConflicts]);
 
-  const filteredSchemaFields = useMemo(() => {
-    if (onlyShowConflicts) return schemaFields.filter((field) => field.type === 'conflict');
-    return schemaFields;
-  }, [onlyShowConflicts, schemaFields]);
+  // update filteredDataTypes as filter by field types are selected
+  const filteredDataTypes = useMemo(() => {
+    const selectedDataTypes = selectedEsFieldTypes
+      .filter((option) => option.checked === 'on')
+      .map((option) => option.label);
+    return selectedDataTypes;
+  }, [selectedEsFieldTypes]);
+
+  const [filteredSchemaFields, totalFieldsWithConflicts] = useMemo(() => {
+    let totalFieldsWithConflict = 0;
+    if (onlyShowConflicts) {
+      const fieldsWithConflicts = schemaFields.filter((field) => field.type === 'conflict');
+      // apply filters on fields with conflicts
+      if (filteredDataTypes.length > 0) {
+        const filteredFieldsWithConflicts = fieldsWithConflicts.filter((field) =>
+          field.indices.some((i) => filteredDataTypes.includes(i.type))
+        );
+        totalFieldsWithConflict = fieldsWithConflicts.length - filteredFieldsWithConflicts.length;
+        return [filteredFieldsWithConflicts, totalFieldsWithConflict];
+      }
+      return [fieldsWithConflicts, totalFieldsWithConflict];
+    }
+    return filteredDataTypes.length > 0
+      ? [
+          schemaFields.filter((field) => filteredDataTypes.includes(field.type)),
+          totalFieldsWithConflict,
+        ]
+      : [schemaFields, totalFieldsWithConflict];
+  }, [onlyShowConflicts, schemaFields, filteredDataTypes]);
```

Comment:
thanks for removing the `set...(` calls from this `useMemo`. i think we can still simplify this.

as i see it we have two filters that we `AND` together. because `AND` is commutative we can perform these operations in sequence and the sequence doesn't matter.

the first filter is if we are only showing conflicts:

```typescript
const maybeFilteredForConflicts = useMemo(() => {
  if (onlyShowConflicts) return schemaFields.filter((field) => field.type === 'conflict');

  return schemaFields
}, [onlyShowConflicts, schemaFields])
```

once we have that we can filter down for any specific types we're looking for

```typescript
const maybeFilteredForSelectedTypes = useMemo(() => {
  if (filteredDataTypes.length === 0) return maybeFilteredForConflicts

  return maybeFilteredForConflicts.filter(field =>
    field.indices.some(i => filteredDataTypes.includes(i.type))
  )
}, [maybeFilteredForConflicts, filteredDataTypes])
```

for clarity maybe we define `filteredSchemaFields` still

```typescript
const filteredSchemaFields = maybeFilteredForSelectedTypes
```

the last thing we need to know is how many conflicts are hidden by the type filters. we only really need this in the UI if we're using both filters.

```typescript
const conflictsHiddenByTypeFilters =
  onlyShowConflicts && filteredDataTypes.length > 0
    ? maybeFilteredForConflicts.length - maybeFilteredForSelectedTypes.length
    : 0
```

```typescript
{conflictsHiddenByTypeFilters > 0 && (
  <EuiCallOut ...
    {/* ... *}
    <FormattedMessage
      values={{ conflictsHiddenByTypeFilters }}
)}
```